### PR TITLE
Web UI

### PR DESCRIPTION
**Pull request by arpankanwer**
In this PR, the front end is created in ReactJs, and it is linked to the controller with the help of an API. There are some features that have been added to this, which are as follows:

1) Weather page: It is linked with the weather API and shows precipitation percentages for the next hours as well as for the current hour. Along with that, we can manually water plants in a specific zone and at a specific time, and we can also mention the time to water each zone. It also shows the current zone watering status in real time from the esp controller and also shows commands that are in queue.

2) Gallery page: It has a slider that shows photos that are clicked in gardening events.

### CODE REVIEW

File: Webpage/webpage/src/styles/style.css
Comment (mcassar4) [inline Webpage/webpage/src/styles/style.css:1]: Lets make the style more representative of the Evertz colour scheme. Black background with yellow lettering

File: Webpage/webpage/src/pages/weather.jsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import axios from 'axios';
+import '../styles/style.css';
+
+const API_BASE_URL = 'http://localhost:5000'; // Update with your server URL
```
Comment (mcassar4) [inline Webpage/webpage/src/pages/weather.jsx:5]: This must be updated with the backend deployed server

File: Webpage/webpage/README.md
Comment (mcassar4) [inline Webpage/webpage/README.md:1]: This should be updated to reflect the instructions to run on the server

Comment (mcassar4) [reply Webpage/webpage/README.md:1]: Also, separate instructions for development

**Review by mcassar4 — DISMISSED** [review]

File: Webpage/webpage/README.md
Comment (arpankanwer) [reply Webpage/webpage/README.md:1]: In this PR, the front end is created in ReactJs, and it is linked to the controller with the help of an API. There are some features that have been added to this, which are as follows:

1) Weather page: It is linked with the weather API and shows precipitation percentages for the next hours as well as for the current hour. Along with that, we can manually water plants in a specific zone and at a specific time, and we can also mention the time to water each zone. It also shows the current zone watering status in real time from the esp controller and also shows commands that are in queue.

2) Gallery page: It has a slider that shows photos that are clicked in gardening events.

**Review by tristanalderson — APPROVED** [review]

**Pull request merged**